module/zfs/zvol.c

```diff
@@ -847,8 +850,10 @@ zvol_discard(void *arg)
 		size = end - start;
 	}
 
-	if (start >= end)
-		goto out;
+	if (start >= end) {
+		error = SET_ERROR(EIO);
```

Comment:
It appears this check was added by commit 089fa91bc51dc10da5a75a6d9fff77f186fccb23 as a optimization to skip expensive non-aligned discards.  We'll want to keep this.